library/core/src/main/java/com/google/android/exoplayer2/text/subrip/SubripDecoder.java

```diff
@@ -108,12 +156,122 @@ protected SubripSubtitle decode(byte[] bytes, int length, boolean reset) {
+   * <p>
+   * Based on SSA v4+ specs the alignment tag can have the following form: {\an[1-9},
+   * where the number specifies the direction (based on the numpad layout).
+   * Note. older SSA scripts may contain tags like {\a1[1-9]} but these are based on
+   * other direction rules, but multiple sources says that these are deprecated, so no support here either
+   *
+   * @param alignmentTag Alignment tag
+   * @return Built cue
+   */
+  private Cue buildCue(Spanned text, String alignmentTag) {
+    // Default values used for positioning the subtitle in case of align tags
+    float line = DEFAULT_END_FRACTION, position = DEFAULT_MID_FRACTION;
+    @Cue.AnchorType int positionAnchor = Cue.ANCHOR_TYPE_MIDDLE;
+    @Cue.AnchorType int lineAnchor = Cue.ANCHOR_TYPE_END;
+
+    switch (alignmentTag) {
+      case ALIGN_BOTTOM_LEFT:
+        line = DEFAULT_END_FRACTION;
+        position = DEFAULT_START_FRACTION;
+        positionAnchor = Cue.ANCHOR_TYPE_START;
+        lineAnchor = Cue.ANCHOR_TYPE_END;
+        break;
+      case ALIGN_BOTTOM_MID:
+        line = DEFAULT_END_FRACTION;
+        position = DEFAULT_MID_FRACTION;
+        positionAnchor = Cue.ANCHOR_TYPE_MIDDLE;
+        lineAnchor = Cue.ANCHOR_TYPE_END;
+        break;
+      case ALIGN_BOTTOM_RIGHT:
+        line = DEFAULT_END_FRACTION;
+        position = DEFAULT_END_FRACTION;
+        positionAnchor = Cue.ANCHOR_TYPE_END;
+        lineAnchor = Cue.ANCHOR_TYPE_END;
+        break;
+      case ALIGN_MID_LEFT:
+        line = DEFAULT_MID_FRACTION;
+        position = DEFAULT_START_FRACTION;
+        positionAnchor = Cue.ANCHOR_TYPE_START;
+        lineAnchor = Cue.ANCHOR_TYPE_MIDDLE;
+        break;
+      case ALIGN_MID_MID:
+        line = DEFAULT_MID_FRACTION;
+        position = DEFAULT_MID_FRACTION;
+        positionAnchor = Cue.ANCHOR_TYPE_MIDDLE;
+        lineAnchor = Cue.ANCHOR_TYPE_MIDDLE;
+        break;
+      case ALIGN_MID_RIGHT:
+        line = DEFAULT_MID_FRACTION;
+        position = DEFAULT_END_FRACTION;
+        positionAnchor = Cue.ANCHOR_TYPE_END;
+        lineAnchor = Cue.ANCHOR_TYPE_MIDDLE;
+        break;
+      case ALIGN_TOP_LEFT:
+        line = DEFAULT_START_FRACTION;
+        position = DEFAULT_START_FRACTION;
+        positionAnchor = Cue.ANCHOR_TYPE_START;
+        lineAnchor = Cue.ANCHOR_TYPE_START;
+        break;
+      case ALIGN_TOP_MID:
+        line = DEFAULT_START_FRACTION;
+        position = DEFAULT_MID_FRACTION;
+        positionAnchor = Cue.ANCHOR_TYPE_MIDDLE;
+        lineAnchor = Cue.ANCHOR_TYPE_START;
+        break;
+      case ALIGN_TOP_RIGHT:
+        line = DEFAULT_START_FRACTION;
+        position = DEFAULT_END_FRACTION;
+        positionAnchor = Cue.ANCHOR_TYPE_END;
+        lineAnchor = Cue.ANCHOR_TYPE_START;
+        break;
+    }
+
+    return new Cue(text, Layout.Alignment.ALIGN_NORMAL, line, Cue.LINE_TYPE_FRACTION, lineAnchor, position, positionAnchor, Cue.DIMEN_UNSET);
```

Comment:
Can you use null for text alignment? `Cue` implies that's what you should use unless you know better.